pkg/routeagent-driver/handlers/kubeproxy-iptables/kp-iptables.go

```diff
@@ -44,31 +44,35 @@ type SyncHandler struct {
 	cleanupHandlers []cleanup.Handler
 }
 
-func NewSyncHandler(env constants.Specification, smClientSet clientset.Interface) *SyncHandler {
-	return &SyncHandler{
-		clusterID:            env.ClusterID,
-		namespace:            env.Namespace,
-		localClusterCidr:     env.ClusterCidr,
-		localServiceCidr:     env.ServiceCidr,
-		localCableDriver:     "",
-		remoteSubnets:        util.NewStringSet(),
-		remoteVTEPs:          util.NewStringSet(),
-		routeCacheGWNode:     util.NewStringSet(),
-		syncHandlerMutex:     &sync.Mutex{},
-		isGatewayNode:        false,
-		wasGatewayPreviously: false,
-		vxlanDevice:          nil,
-		vxlanGwIP:            nil,
-		smClientSet:          smClientSet,
-	}
+func NewSyncHandler(env constants.Specification, smClientSet clientset.Interface) event.Handler {
```

Comment:
As discussed over Slack, IMHO GetNetworkPlugin -> GetNetworkPlugins() might be more simpler.
Also, xfrm specific fixes are required even in the Generic driver (as we should not delete the xfrm rules which are not programmed by route-agent).

CC: @tpantelis @skitt

Reply:
done